src/connection.ts

```diff
@@ -3129,22 +3134,21 @@ class Connection extends EventEmitter {
 
       let message: Message;
 
-      request.once('cancel', () => {
-        if (message.writable) {
-          // - if the message is still writable, we'll set the ignore bit
-          //   and end the message.
-          message.ignore = true;
-          message.end();
+      const onCancel = () => {
+        // set the ignore bit and end the message.
+        message.ignore = true;
+        message.end();
 
-          this.clearRequestTimer();
-          this.createCancelTimer();
+        this.clearRequestTimer();
+        this.createCancelTimer();
 
-          if (request instanceof Request && request.paused) {
-            // resume the request if it was paused so we can read the remaining tokens
-            request.resume();
-          }
+        if (request instanceof Request && request.paused) {
+          // resume the request if it was paused so we can read the remaining tokens
+          request.resume();
         }
-      });
```

Comment:
We no longer need the `message.writable` check here, because a message is guaranteed to be writable until the  `finish` event has been called, and that's also when the event handler will be removed.